### ISSUE TITLE
Save an image or other node entity in assayreg

### ISSUE THREAD

**Issue by strets123** [thesgc/chembiohub_helpdesk#543]
Uploading an image or group of images will be a required feature to allow a number of users to push their  phenotypic  assay data. We have also seen users wanting to manage long matrices of cell lines with slightly different variations.

Uploading an image is conceptually similar to uploading a compound.

For example, images often need to be uploaded in a separate file to the data about the images.

Compounds are sometimes uploaded separately to their metadata.

Both require a thumbnail and a gallery view.

Images would often be uploaded in bulk as a zip/tar/bz archive file
Compounds would also often be loaded in bulk.

Linking an assay datapoint to its image or spectrum is very conceptually similar to linking an assay to its compound.

Therefore I propose that the internet explorer issue with chemreg could be solved by refactoring chemreg to push data to assayreg datastore, with the chemicals stored in the same way images would be stored.

Cross linking the two things would then be dealt with by the same code that would allow a multi image upload to be linked to its datapoints,

This code would also allow cell lines etc to be registered.

**Comment by strets123** [comment]
It should be noted that currently the definition of 10, l1, 12 and 13 are hard-coded into the search interface. Arguably there will be different requirements for cell line search compared to assay search. And agian different requirements for image sselector but the base data they work from is the same.

We have the possibility of varying the template on a feature by feature basis.

In order to do this there needs to be a simple definition of a feature at the server level that looks something like this:

```
[
    {
        "name": "Assays",
        "allowed_data_types": {
            "l0": {
                "name": "Project",
                "lookups": null
            },
            "l1": {
                "name": "Sub-Project",
                "lookups": null
            },
            "l2": {
                "name": "Assay",
                "lookups": null
            },
            "l3": {
                "name": "Activity",
                "lookups": null
            }
        },
            "search": {"controller": "assaySearchController", "template": ""}
            "overview": {"controller": "overviewController" ,"template": ""}
    },
    {
        "name": "Compounds",
        "allowed_data_types": {
            "l0": {
                "name": "Project",
                "lookups": null
            },
            "l1": {
                "name": "Compounds",
                "lookups": null
            },
            "l2": {
                "name": "Compound Batches",
                "lookups": null
            },
            "search": {"controller": "compoundSearchController", "template": ""}
            "overview": {"controller": "overviewController" ,"template": "compound_overview.html"}
        }
    }
]
```

**Comment by PaulBarrett79** [comment]
Solved by #585
